transport_analysis/viscosity.py

```diff
@@ -0,0 +1,191 @@
+
+        # local
+        self.atomgroup = atomgroup
+        self.n_particles = len(self.atomgroup)
+
+    def _prepare(self):
+        """
+        Set up viscosity, mass, velocity, and position arrays
+        before the analysis loop begins
+        """
+        # 2D viscosity array of frames x particles
+        self.results.visc_by_particle = np.zeros(
+            (self.n_frames, self.n_particles)
+        )
+
+        self._mass_array = self.atomgroup.masses
+        self._mass_reshape = self._mass_array.reshape(
+            1, len(self._mass_array, 1)
+        )
+
+        # 3D arrays of frames x particles x dimensionality
+        # for velocities and positions
+        self._velocity_array = np.zeros(
+            (self.n_frames, self.n_particles, self.dim_fac)
+        )
+
+        self._position_array = np.zeros(
+            (self.n_frames, self.n_particles, self.dim_fac)
+        )
+        # self.results.timeseries not set here
+
+    def _parse_dim_type(self):
+        r"""Sets up the desired dimensionality of the calculation."""
+        keys = {
+            "x": [0],
+            "y": [1],
+            "z": [2],
+            "xy": [0, 1],
+            "xz": [0, 2],
+            "yz": [1, 2],
+            "xyz": [0, 1, 2],
+        }
+
+        self.dim_type = self.dim_type.lower()
+
+        try:
+            self._dim = keys[self.dim_type]
+        except KeyError:
+            raise ValueError(
+                "invalid dim_type: {} specified, please specify one of xyz, "
+                "xy, xz, yz, x, y, z".format(self.dim_type)
+            )
+
+        self.dim_fac = len(self._dim)
+
+    def _single_frame(self):
+        """
+        Constructs arrays of velocities and positions
+        for viscosity computation.
+        """
+        # This runs once for each frame of the trajectory
+
+        # The trajectory positions update automatically
+        # You can access the frame number using self._frame_index
+
+        # trajectory must have velocity information
+        if not (self._ts.has_velocities and self._ts.has_positions):
+            raise NoDataError(
+                "Helfand viscosity computation requires "
+                "velocities and positions in the trajectory"
+            )
+
+        # set shape of velocity array
+        self._velocity_array[self._frame_index] = self.atomgroup.velocities[
+            :, self._dim
+        ]
+
+        # set shape of position array
+        self._position_array[self._frame_index] = self.atomgroup.positions[
+            :, self._dim
+        ]
+
+    def _conclude(self):
+        r"""Calculates viscosity via the simple "windowed" algorithm."""
+        lagtimes = np.arange(1, self.n_frames)
+
+        # improve precision with np.float64
+        masses = self._mass_reshape.astype(np.float64)
+        velocities = self._velocity_array.astype(np.float64)
+        positions = self._position_array.astype(np.float64)
```

Comment:
I believe that the default float in python is already 64 bits, what will this do?